File: src/quark/designdocs/orders.design/views/queue/map.cpp
/*
 * map.cpp
 *
 *  Created on: Jun 4, 2017
 *      Author: ondra
 */

#include <couchcpp/api.h>

void mapdoc(Document doc) {

	if (!doc["_deleted"].getBool()) {

		StrViewA name = doc.getID();
		if (name == "settings" || name == "error") {
			emit(nullptr,nullptr);
		}
		if (name.substr(0,2) == "o." && doc["finished"].getBool() == false) {
			emit(true,nullptr);
		}
	}


}

File: src/quark/designdocs/orders.design/views/queue/map.js
function(doc) {
	
	if (!doc._deleted) {		
		if (doc._id == "error") {
			emit(1,null);
		} else if (doc._id == "settings") {
			emit(2,null);
		} else if (doc._id.substr(0,2) == "o." && !doc.finished){
			emit(doc.timeModified, null);
		}		
	}
	
	
}